dynamic-plugin/src/hooks/useDashboards.ts

```diff
@@ -98,9 +98,54 @@ export function useDashboards(dashboardId?: string) {
           setWidgets((prev) => [...prev, ...(generateUIResponse.widgets ?? [])]);
         }
       }
+
+      // If chatbot updated dashboard metadata, refetch the active dashboard to reflect changes
+      if (
+        toolName === 'set_dashboard_metadata'
+      ) {
+        (async () => {
+          try {
+            const resp = dashboardId
+              ? await dashboardClient.current.getDashboard(dashboardId)
+              : await dashboardClient.current.getActiveDashboard();
+            if (resp) {
+              const normalizedActive = DashboardUtils.normalizeResponse(resp as any);
+              setActiveDashboard(normalizedActive);
+              setWidgets(normalizedActive?.widgets ?? []);
+            }
+          } catch (e) {
+            console.error('Error refetching after metadata update:', e);
+          }
+        })();
+      }
     });
   }
 
+   // Listen for manual refresh events (e.g., after inline save) and refetch active dashboard
+   useEffect(() => {
+    async function refetchActive() {
```

Comment:
useDashboard can expose the state setters and you can just update it when you get the response in the onConfirm callbacks to avoid an additional network call.